src/components/SearchBox.js

```diff
@@ -87,12 +92,24 @@ class SearchBox extends Component {
         isOpen: false
       });
     }
+    if (analytics && prevProps.analyticsConfig !== analyticsConfig)
```

Comment:
@ShahAnuj2610 You need to compare the inner values too, you can use `JSON.stringify` for the comparison.